src/main/java/ai/knowly/langtorch/preprocessing/splitter/text/TextSplitter.java

```diff
@@ -0,0 +1,136 @@
+package ai.knowly.langtorch.preprocessing.splitter.text;
+
+import ai.knowly.langtorch.schema.io.DomainDocument;
+import ai.knowly.langtorch.schema.io.Metadata;
+import org.apache.commons.collections4.keyvalue.MultiKey;
+import org.apache.commons.collections4.map.MultiKeyMap;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * The TextSplitter class provides functionality for splitting text into chunks.
+ */
+public abstract class TextSplitter {
+
+    public final int chunkSize;
+
+    public final int chunkOverlap;
+
+    public TextSplitter(int chunkSize, int chunkOverlap) {
+        this.chunkSize = chunkSize;
+        this.chunkOverlap = chunkOverlap;
+        if (this.chunkOverlap >= this.chunkSize) {
+            throw new IllegalArgumentException("chunkOverlap cannot be equal to or greater than chunkSize");
+        }
+    }
+
+    abstract public List<String> splitText(String text);
+
+    public List<DomainDocument> createDocuments(List<String> texts, Optional<List<Metadata>> docMetadatas) {
```

Comment:
are you imagining something recursive here or that we still return a list of documents from one input? I think both can make sense because even if we're only passing single text and metadata we can still expect the text splitter to return multiple docs

Reply:
I've renamed this method `createDocumentsSplitFromList`, have created another method called `createDocumentsSplitFromSingle` and a private method `addDocumentFromWordChunk`